homeassistant/components/synology_dsm/sensor.py

```diff
@@ -55,106 +49,23 @@ async def async_setup_entry(
         for disk in entry.data.get(CONF_DISKS, api.storage.disks_ids):
-            sensors += [
-                SynoNasStorageSensor(
+            entities += [
+                SynoDSMStorageSensor(
                     api, sensor_type, STORAGE_DISK_SENSORS[sensor_type], disk
                 )
                 for sensor_type in STORAGE_DISK_SENSORS
             ]
 
-    async_add_entities(sensors)
-
-
-class SynoNasSensor(Entity):
-    """Representation of a Synology NAS sensor."""
-
-    def __init__(
-        self,
-        api: SynoApi,
-        sensor_type: str,
-        sensor_info: Dict[str, str],
-        monitored_device: str = None,
-    ):
-        """Initialize the sensor."""
-        self._api = api
-        self.sensor_type = sensor_type
-        self._name = f"{BASE_NAME} {sensor_info[0]}"
-        self._unit = sensor_info[1]
-        self._icon = sensor_info[2]
-        self.monitored_device = monitored_device
-        self._unique_id = f"{self._api.information.serial}_{sensor_info[0]}"
-
-        if self.monitored_device:
-            self._name += f" ({self.monitored_device})"
-            self._unique_id += f"_{self.monitored_device}"
-
-        self._unsub_dispatcher = None
-
-    @property
-    def unique_id(self) -> str:
-        """Return a unique ID."""
-        return self._unique_id
-
-    @property
-    def name(self) -> str:
-        """Return the name."""
-        return self._name
-
-    @property
-    def icon(self) -> str:
-        """Return the icon."""
-        return self._icon
-
-    @property
-    def unit_of_measurement(self) -> str:
-        """Return the unit the value is expressed in."""
-        if self.sensor_type in TEMP_SENSORS_KEYS:
-            return self._api.temp_unit
-        return self._unit
-
-    @property
-    def device_state_attributes(self) -> Dict[str, any]:
-        """Return the state attributes."""
-        return {ATTR_ATTRIBUTION: ATTRIBUTION}
-
-    @property
-    def device_info(self) -> Dict[str, any]:
-        """Return the device information."""
-        return {
-            "identifiers": {(DOMAIN, self._api.information.serial)},
-            "name": "Synology NAS",
-            "manufacturer": "Synology",
-            "model": self._api.information.model,
-            "sw_version": self._api.information.version_string,
-        }
-
-    @property
-    def should_poll(self) -> bool:
-        """No polling needed."""
-        return False
-
-    async def async_update(self):
-        """Only used by the generic entity update service."""
-        await self._api.update()
-
-    async def async_added_to_hass(self):
-        """Register state update callback."""
-        self._unsub_dispatcher = async_dispatcher_connect(
-            self.hass, self._api.signal_sensor_update, self.async_write_ha_state
-        )
-
-    async def async_will_remove_from_hass(self):
-        """Clean up after entity before removal."""
-        self._unsub_dispatcher()
```

Comment:
We should migrate the old unique ids so this change isn't a breaking change.

There are two ways to do this.

Examples in

1. `powerwall/__init__.py` and `coronavirus/__init__.py` 
2. `sense/binary_sensor.py`

Reply:
Done